pallets/avn/src/tests/session_handler_tests.rs

```diff
@@ -1,167 +1,272 @@
+                    TestAccount::derive_validator(5),
+                    TestAccount::derive_validator(3),
+                    TestAccount::derive_validator(4),
+                    TestAccount::derive_validator(1),
+                    TestAccount::derive_validator(2),
+                ]);
+            })
+        }
+
+        #[test]
+        fn then_avn_does_not_know_collator(){
+            let mut ext = setup_initial_collators();
+
+            ext.execute_with(|| {
+                add_two_collators_and_force_two_sessions();
+
+                let current_collators = avn_known_collators();
+                assert_eq!(current_collators, vec![
+                    TestAccount::derive_validator(5),
+                    TestAccount::derive_validator(3),
+                    TestAccount::derive_validator(4),
+                    TestAccount::derive_validator(1),
+                    TestAccount::derive_validator(2),
+                ]);
+
+                remove_collator_candidate(TestAccount::derive_validator(5).account_id, 5);
+
+                advance_session();
+                advance_session();
+
+                let final_collators = avn_known_collators();
+                assert_eq!(final_collators, vec![
+                    TestAccount::derive_validator(3),
+                    TestAccount::derive_validator(4),
+                    TestAccount::derive_validator(1),
+                    TestAccount::derive_validator(2),
+                ]);
+            })
+        }
+    }
+
+    mod when_more_than_desired_candidates_register {
+        use super::*;
+
+        fn setup_adds_seven_collators() {
+            for id in 4u64..10u64 {
+                super::add_collator_candidate(TestAccount::derive_validator(id).account_id, id);
+            }
+        }
+
+        #[test]
+        fn then_no_change_visible_in_following_session(){
+            let mut ext = setup_initial_collators();
+            ext.execute_with(|| {
+                let initial_collators = avn_known_collators();
+                setup_adds_seven_collators();
+
+                advance_session();
+
+                let final_collators = avn_known_collators();
+
+                assert_eq!(initial_collators, final_collators);
+            })
+        }
+
+        #[test]
+        fn then_after_two_sessions_avn_knows_subset_of_new_candidates(){
+            let mut ext = setup_initial_collators();
+            ext.execute_with(|| {
+                setup_adds_seven_collators();
+
+                advance_session();
+                advance_session();
+
+                let final_collators = avn_known_collators();
+
+                assert_eq!(final_collators, vec![
+                    TestAccount::derive_validator(5),
+                    TestAccount::derive_validator(3),
+                    TestAccount::derive_validator(4),
+                    TestAccount::derive_validator(1),
+                    TestAccount::derive_validator(2),
+                ]);
```

Comment:
This is not what I'd expect. How do we know the limit of collators that can join? Where do we specify it? How can we guarantee this order?

Reply:
A sort_collators function has been implemented to fix this problem.